Change box colliders to circle colliders

levyadams opened the pull request:
These circle colliders are bad and work badly but better than the box colliders for what we need. Later we will need to really dial each character and projectile in to work for balance.

ignacioxd commented on game/src/js/objects/characters/Character.js:
```diff
@@ -50,6 +51,8 @@ export default class Character extends Phaser.Physics.Arcade.Sprite {
     this.setVelocity(0, 0);
     let projectile = new Projectile(this.scene, this.x, this.y, this.projectileType, targetX, targetY);
     this.scene.projectiles.add(projectile);
+    projectile.body.isCircle = true;
```
I'd move this to the constructor of the Projectile class.

ignacioxd replied:
Remember to remove this from here also.

ignacioxd commented on game/src/js/objects/characters/Character.js:
```diff
@@ -50,6 +51,8 @@ export default class Character extends Phaser.Physics.Arcade.Sprite {
     let projectile = new Projectile(this.scene, this.x, this.y, this.projectileType, targetX, targetY);
     this.scene.projectiles.add(projectile);
+    projectile.body.isCircle = true;
+    projectile.body.setCircle(40,-5,-5);
```
I wonder how well this circle will work for other types of projectiles. Have you tested it with anything other than 'orb' by any chance?

levyadams commented:
Switched hard-coded area's out for constructor class magic. Added function inside of projectile.js to switch the size of the collider depending on the name passed from the character.projectileType string class as well.

ignacioxd commented on game/src/js/objects/characters/FireMonster.js:
```diff
@@ -1,8 +1,8 @@
 export default class FireMonster extends Character {
-  constructor(scene, x, y) {
-    super(scene, x, y, 'fire_monster');
+  constructor(scene, x, y,colSize,offsetX,offsetY) {
```
Since the colliders will probably be the same each time an instance of a character is created, these values should not be passed in when the character subclasses (e.g. Priest, Knight, etc.) are constructed. Instead, just put the respective values here in the `super()` method. For example,

```javascript
constructor(scene, x, y) {
  super(scene, x, y, 70, 95, 65);
  ...
}
```

ignacioxd commented:
Or even better, still accept the parameters on the constructors, but set a default value for them:

```javascript
constructor(scene, x, y, colSize = 70, offsetX = 95, offsetY = 65) {
  super(scene, x, y, colSize, offsetX, offsetY);
  ...
}
```

levyadams commented:
Removed explicit call from subclass constructor, now handled inside of the subclass itself. Removed Projectile.IsCircle() from character.fire() as function.setCircle() does the same thing inside of the projectile constructor.
